dftimewolf/lib/state.py

```diff
@@ -60,6 +63,27 @@ def __init__(self, config: Type[Config]) -> None:
     self.streaming_callbacks = {}  # type: Dict[Type[interface.AttributeContainer], List[Callable[[Any], Any]]]  # pylint: disable=line-too-long
     self._abort_execution = False
 
+  def __deepcopy__(self, memo: Dict[Any, Any]) -> object:
+    """Deepcopy override. This creates a new copy of the state object. This is
+    needed because we copy ThreadAwareModules for parallel processing, and they
+    have a reference to the state."""
```

Comment:
The deepcopy was necessary because the thread lock in the global state cannot be deepcopy'd - So the deepcopy override creates an individual thread lock.

Reply:
Removed with updated container handling.